### ISSUE TITLE
Memory leak in example project

### ISSUE THREAD

**Issue by b00giZm** [artsy/ezel#43]
You might want to review this part of your example project:

```javascript
var Commits = require('../../collections/commits');

exports.index = function(req, res, next) {
  var commits = new Commits(null, {
    owner: 'artsy',
    repo: 'flare'
  });
  commits.fetch({
    success: function() {
      res.locals.sd.COMMITS = commits.toJSON();
      res.render('index', { commits: commits.models });
    },
    error: function(m, err) { next(err.text); }
  });
};
```

We followed your example and stumbled across some serious problems concerning memory consumption for high traffic apps. Creating Backbone collections for every request and using them to fetch (and rendering) data seems to be problematic, because the GC won't clean them up and the Node process will allocate more and more memory.

![memory](https://cloud.githubusercontent.com/assets/59077/4388599/1c113e32-43ed-11e4-9826-079e54212f87.png)

I haven't yet fully understood if this is an Backbone issue or maybe caused by your backbone-super-sync module.

**Comment by craigspaeth** [comment]
So I believe I traced down the main cause of the memory leak in [backbone-super-sync](https://github.com/artsy/backbone-super-sync)
. Turns out Backbone.sync is, unsurprisingly, an easy bottle neck for leaky memory. After a lot of trial and error I came to the conclusion that a combination of forcing timeouts on requests and refactoring out closures solved the issue—at least for various applications I tested on. Here's the code for details https://github.com/artsy/backbone-super-sync/pull/33

Right now one of our Ezel applications has a flat memory graph.

![image](https://cloud.githubusercontent.com/assets/555859/8621501/f1804876-26f3-11e5-971d-e53ab3d64711.png)

Another has very spikey memory use but it appears to at least be freeing large chunks of memory. (My hunch for the cause of this is we have some routes where we load a large amount of data into Backbone collections (e.g. iterates though an API endpoint that returns paginated array data and we "fetch until end" loading thousands of entities in memory).

![image](https://cloud.githubusercontent.com/assets/555859/8621543/236d8eb6-26f4-11e5-8419-a7bc3aac44f9.png)

I tested this on the example project in Ezel and pinged it at ~4k rpm on a 512mb heroku instance and while performance would degrade (probably b/c of heavy GCing) it did hover just under the max RAM.

So :pray: the _leak_ is fixed here, but there's still probably more wins to be had in terms of memory hungriness. Any thoughts and comments of course are much appreciated! I'll call this one closed for now.
